models/pull.go

```diff
@@ -272,6 +272,28 @@ const (
 	MergeStyleSquash MergeStyle = "squash"
 )
 
+// CheckUserAllowedToMerge checks whether the user is allowed to merge
+func (pr *PullRequest) CheckUserAllowedToMerge(doer *User) (err error) {
+	if doer == nil {
+		return ErrNotAllowedToMerge{
+			"Not signed in",
+		}
+	}
+	if err = pr.GetBaseRepo(); err != nil {
```

Comment:
No need to load repository if it is already loaded (need check `if pr.BaseRepo == nil {`)

Reply:
I added it, but GetBaseRepo() already checks for this:
```go
func (pr *PullRequest) GetBaseRepo() (err error) {
	if pr.BaseRepo != nil {
		return nil
	}
```

Reply:
Sorry I overlooked that function than... I was checking code but somehow missed that check. If it does than yes there is no need to do additional check